lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -196,7 +196,8 @@ deriveEquatable_enum_eq(TypeChecker &tc, Decl *parentDecl, EnumDecl *enumDecl) {
   //   case A, B, C
   //
   //   @derived
-  //   func ==(a: SomeEnum<T...>, b: SomeEnum<T...>) -> Bool {
+  //   @_implements(Equatable, ==(_:_:))
+  //   func _DerivedEnumEquals(a: SomeEnum<T...>, b: SomeEnum<T...>) -> Bool {
```

Comment:
Is this going to look ugly in generated interface printing? Maybe we could special case that to print this as == (or just skip synthesized decls altogether).